flake.nix

```diff
@@ -0,0 +1,51 @@
+{
+  description = "A flake for s2n-tls";
+
+  inputs.nixpkgs.url = github:NixOS/nixpkgs/nixos-22.11;
+
+  outputs = { self, nix, nixpkgs, flake-utils }: 
+     flake-utils.lib.eachDefaultSystem (system:
+      let pkgs = nixpkgs.legacyPackages.${system};
+   in rec {
+      packages.s2n-tls = pkgs.stdenv.mkDerivation {
+        src = self;
+        name = "s2n-tls";
+        inherit system; 
+        # TODO: override the s2n-tls version
+
+        buildInputs = [ pkgs.cmake
+                        pkgs.openssl_3 ]; # s2n-config has find_dependency LibCrypto
+
+        cmakeFlags = [
+            "-DBUILD_SHARED_LIBS=ON"
+            "-DUNSAFE_TREAT_WARNINGS_AS_ERRORS=OFF" # disable -Werror
+            "-DCMAKE_BUILD_TYPE=RelWithDebInfo"
+            "-DS2N_NO_PQ=1" # TODO: set when system like aarch64/mips,etc
+        ];
+
+      };
+      defaultPackage = packages.s2n-tls;
+      packages.s2n-tls-openssl3 = pkgs.s2n-tls.overrideAttrs (finalAttrs: previousAttrs: {
+        doCheck = true;
+      });
+      packages.s2n-tls-openssl11 = pkgs.s2n-tls.overrideAttrs (finalAttrs: previousAttrs: {
+        doCheck = true;
+        buildInputs = [ pkgs.openssl_1_1 ];
```

Comment:
For my edification: how does this work? If buildInputs is overridden isn't the cmake dependency also being overridden? Are we implicitly relying on cmake?

Reply:
I wasn't able to find a specific call out in the documentation if the `overrideAttrs` was destructive or an append to the list...but it appears that openssl3 is always available as a fallback (see my comments about some of the libcryptos no installing a library).